packages/navigator/src/browser/navigator-widget.tsx

```diff
@@ -139,6 +143,36 @@ export class FileNavigatorWidget extends FileTreeWidget {
         return super.renderTree(model);
     }
 
+    protected renderTailDecorations(node: TreeNode, props: NodeProps): React.ReactNode {
+        const tailDecorations = this.getDecorationData(node, 'tailDecorations').filter(notEmpty).reduce((acc, current) => acc.concat(current), []);
+
+        if (tailDecorations.length === 0) {
+            return;
+        }
+
+        if (CompositeTreeNode.is(node)) {
+            return this.renderTailDecorationsForCompositeNode(node, props, tailDecorations);
+        } else {
+            return this.renderTailDecorationsForNode(node, props, tailDecorations);
+        }
+    }
```

Comment:
We can make use of `FileStatNode` and rename the composite method:

```suggestion
        if (FileStatNode.is(node) && node.fileState.isDirectory) {
            return this.renderTailDecorationsForFolderNode(node, props, tailDecorations);
        } else {
            return this.renderTailDecorationsForNode(node, props, tailDecorations);
        }
    }
```